src/node_api_helpers.h

```diff
@@ -38,6 +45,289 @@
+    napi_env env = napi_get_current_env();
+    return napi_get_type_of_value(env, val) == napi_regexp;
+  }
+  inline bool IsDate(napi_value val) {
+    napi_env env = napi_get_current_env();
+    return napi_get_type_of_value(env, val) == napi_date;
+  }
+  inline bool IsNull(napi_value val) {
+    napi_env env = napi_get_current_env();
+    return napi_get_type_of_value(env, val) == napi_null;
+  }
+
+  inline bool HasInstance(napi_value tpl, napi_value obj) {
+    napi_env env = napi_get_current_env();
+    return napi_instanceof(env, obj, tpl);
+  }
+
+  inline napi_value CopyBuffer(const char* buf, uint32_t size) {
+    napi_env env = napi_get_current_env();
+    return napi_buffer_copy(env, buf, size);
+  }
+
+  inline napi_value StringConcat(napi_value str1, napi_value str2) {
+    napi_env env = napi_get_current_env();
+    return napi_string_concat(env, str1, str2);
+  }
+
+  inline napi_value MakeCallback(
+      napi_value target,
+      napi_value func,
+      int argc,
+      napi_value* argv
+  ) {
+    napi_env env = napi_get_current_env();
+    return napi_make_callback(env, target, func, argc, argv);
+  }
+
+  // Length helpers
+  inline int Length(napi_value val) {
+    napi_env env = napi_get_current_env();
+    assert(napi_get_type_of_value(env, val) == napi_array);
+    return napi_get_array_length(env, val);
+  }
+
+  inline int Length(napi_callback_info info) {
+    napi_env env = napi_get_current_env();
+    return napi_get_cb_args_length(env, info);
+  }
+
+  // Conversion helpers
+  inline void To(napi_value v, int32_t& o) {
+    napi_env env = napi_get_current_env();
+    o = napi_get_value_int32(env, v);
+  }
+
+  inline void To(napi_value v, uint32_t& o) {
+    napi_env env = napi_get_current_env();
+    o = napi_get_value_uint32(env, v);
+  }
+
+  inline void To(napi_value v, int64_t& o) {
+    napi_env env = napi_get_current_env();
+    o = napi_get_value_int64(env, v);
+  }
+
+  inline void To(napi_value v, bool& o) {
+    napi_env env = napi_get_current_env();
+    o = napi_get_value_bool(env, v);
+  }
+
+  inline void To(napi_value v, double& o) {
+    napi_env env = napi_get_current_env();
+    o = napi_get_number_from_value (env, v);
+  }
+
+  template<typename T>
+  inline T To(napi_value v) {
+    T ret;
+    To(v, ret);
+    return ret;
+  }
+
+  inline napi_value ToString(napi_value v) {
+    napi_env env = napi_get_current_env();
+    return napi_coerce_to_string(env, v);
+  }
+  inline napi_value ToObject(napi_value v) {
+    napi_env env = napi_get_current_env();
+    return napi_coerce_to_object(env, v);
+  }
+
+  inline bool Equals(napi_value lhs, napi_value rhs) {
+    napi_env env = napi_get_current_env();
+    return napi_strict_equals(env, lhs, rhs);
+  }
+
+  inline napi_value* GetArguments(napi_callback_info info) {
+    napi_env env = napi_get_current_env();
+    int len = napi_get_cb_args_length(env, info);
+    napi_value* args = (napi_value*) malloc(len * sizeof(napi_value));
```

Comment:
When does this memory get freed?

It's OK for the helpers to use C++, right? So could this return something like a `std::vector` to make memory-management easier?